src/flexmeasures_client/client.py

```diff
@@ -85,6 +86,19 @@ def __post_init__(self):
             )
         if len(self.password) < 1:
             raise EmptyPasswordError("password cannot be empty")
+        self.determine_port()
+
+    def determine_port(self):
+        parts = self.host.split(':')
+        if len(parts) > 1:
+            if self.port is not None:
+                raise WrongHostError(
+                    f"Cannot set port={self.port} and also as part of host={self.host}"
+                )
+            self.port = int(parts[1])
+        elif self.port is None:
+            self.port = 443 if self.scheme == 'https' else 80
+        self.host = parts[0]
```

Comment:
I would prefer if this line goes in the `if` block, after line 98.
If there is no port in the host string, this function should not touch it.